wolfcrypt/src/asn.c

```diff
@@ -11250,6 +11253,94 @@ static int wc_SetCert_LoadDer(Cert* cert, const byte* der, word32 derSz)
     return ret;
 }
 
+#ifdef WOLFSSL_ALT_NAMES
+
+/* AltName structure contains a sequence of names.
+   WC_ALTNAMESOFFSET is the max header size */
+#define WC_ALTNAMESOFFSET (MAX_SEQ_SZ) 
+
+int wc_CertAddAltName_ex(Cert *cert, byte *name, int nameSz, 
+						 cert_altname_type type) {
+	/* Append encoded value from name into cert->altNames */
+	byte *buf;
+	int i;
+	
+	if (cert == NULL || name == NULL) {
+		return BAD_FUNC_ARG;
+	}
+	if (cert->altNamesSz == 0) {
+		/* Reserve space for header structure */
+		cert->altNamesSz = WC_ALTNAMESOFFSET;
+	}
+	if (cert->altNamesSz + nameSz + MAX_OCTET_STR_SZ > WC_CTC_MAX_ALT_SIZE)
+		return BUFFER_E;		/* Input too large for buffer */
+	if (type < WC_ALTNAME_OTHERNAME || type > WC_ALTNAME_REGISTEREDID) 
+		return BAD_FUNC_ARG;
+	
+	buf = cert->altNames;  
+	i = cert->altNamesSz;		/* Append new name at end of buffer */
+
+	i += SetImplicit(ASN_OCTET_STRING, type, nameSz, buf + i);
+	XMEMCPY(buf + i,name,nameSz);
+	cert->altNamesSz = i + nameSz;
+	return 0;
+}
+
+int wc_CertAddAltNameStr(Cert *cert, byte *name, cert_altname_type type) {
+    /* Call this function N times to add N AltName strings */
+	int nameSz;
+	int rv;
+
+	if (cert == NULL || name == NULL) {
+		rv = BAD_FUNC_ARG;
+	} else if (type == WC_ALTNAME_RFC822NAME || type == WC_ALTNAME_DNSNAME || 
+		type==WC_ALTNAME_URI) {
+			nameSz = XSTRLEN(name);
```

Comment:
Compile error:
```
In file included from ./wolfssl/wolfcrypt/asn.h:29:0,
                 from wolfcrypt/src/asn.c:61:
wolfcrypt/src/asn.c: In function ‘wc_CertAddAltNameStr’:
./wolfssl/wolfcrypt/types.h:428:42: error: pointer targets in passing argument 1 of ‘strlen’ differ in signedness [-Werror=pointer-sign]
         #define XSTRLEN(s1)       strlen((s1))
                                          ^
wolfcrypt/src/asn.c:11298:13: note: in expansion of macro ‘XSTRLEN’
    nameSz = XSTRLEN(name);
             ^~~~~~~
In file included from ./wolfssl/wolfcrypt/types.h:422:0,
                 from ./wolfssl/wolfcrypt/asn.h:29,
                 from wolfcrypt/src/asn.c:61:
/usr/include/string.h:384:15: note: expected ‘const char *’ but argument is of type ‘byte * {aka unsigned char *}’
 extern size_t strlen (const char *__s)
               ^~~~~~
```

Reply:
Again, you must have different compiler settings from me to flag this.  I could add "(char *)" in front of the reference, and that should eliminate the error.